### ISSUE TITLE
Training does not start

### ISSUE THREAD

**Issue by abhoi** [wuhuikai/FastFCN#8]
Hi, I tried running the train.py script for both PContext and ADE20k datasets after following all the instructions. (setup.py, prepare_ade20k.py/prepare_pcontext.py, details-api). However, the training does not start or at least I can't see any progress. For context, I'm running this on Horovod docker (https://github.com/horovod/horovod/blob/master/Dockerfile). I only get the following log and nothing more:

root@ml:/path/# CUDA_VISIBLE_DEVICES=0,1,2,3 python train.py --dataset ade20k --model encnet --jpu --aux --se-loss --backbone resnet50 --checkname encnet_res50_ade20k_train
Namespace(aux=True, aux_weight=0.2, backbone='resnet50', base_size=520, batch_size=128, checkname='encnet_res50_ade20k_train', crop_size=480, cuda=True, dataset='ade20k', dilated=False, epochs=120, ft=False, jpu=True, lateral=False, lr=0.08, lr_scheduler='poly', mode='testval', model='encnet', model_zoo=None, momentum=0.9, ms=False, no_cuda=False, no_val=False, resume=None, save_folder='results', se_loss=True, se_weight=0.2, seed=1, split='val', start_epoch=0, test_batch_size=128, train_split='train', weight_decay=0.0001, workers=16)
BaseDataset: base_size 520, crop_size 480
len(img_paths): 20210
/usr/local/lib/python3.5/dist-packages/torch/nn/_reduction.py:49: UserWarning: size_average and reduce args will be deprecated, please use reduction='mean' instead.
  warnings.warn(warning.format(ret))
Using poly LR Scheduler!
Starting Epoch: 0
Total Epoches: 120
  0%|                                                                                          | 0/157 [00:00<?, ?it/s]Training at 0

=>Epoches 0, learning rate = 0.0800,                 previous best = 0.0000

ml:436:436 [3] misc/ibvwrap.cu:63 NCCL WARN Failed to open libibverbs.so[.1]
ml:436:436 [3] NCCL INFO Using internal Network Socket
NCCL version 2.3.7+cuda9.0
ml:436:436 [3] NCCL INFO nranks 4
ml:436:436 [0] NCCL INFO comm 0x188aaea80 rank 0 nranks 4
ml:436:436 [1] NCCL INFO comm 0x188ab0b80 rank 1 nranks 4
ml:436:436 [2] NCCL INFO comm 0x188aa8f10 rank 2 nranks 4
ml:436:436 [3] NCCL INFO comm 0x188aab010 rank 3 nranks 4
ml:436:436 [0] NCCL INFO NET : Using interface lo:127.0.0.1<0>
ml:436:436 [0] NCCL INFO NET : Using interface eno49:10.102.140.115<0>
ml:436:436 [0] NCCL INFO NET/Socket : 2 interfaces found
ml:436:436 [0] NCCL INFO Could not find real path of /sys/class/net/lo/device
ml:436:436 [0] NCCL INFO CUDA Dev 0, IP Interfaces : lo(SOC) eno49(PHB)
ml:436:436 [1] NCCL INFO Could not find real path of /sys/class/net/lo/device
ml:436:436 [1] NCCL INFO CUDA Dev 1, IP Interfaces : lo(SOC) eno49(PHB)
ml:436:436 [2] NCCL INFO Could not find real path of /sys/class/net/lo/device
ml:436:436 [2] NCCL INFO CUDA Dev 2, IP Interfaces : lo(SOC) eno49(SOC)
ml:436:436 [3] NCCL INFO Could not find real path of /sys/class/net/lo/device
ml:436:436 [3] NCCL INFO CUDA Dev 3, IP Interfaces : lo(SOC) eno49(SOC)
ml:436:436 [3] NCCL INFO Using 256 threads
ml:436:436 [3] NCCL INFO Min Comp Cap 5
ml:436:436 [3] NCCL INFO Ring 00 :    0   1   2   3
ml:436:436 [0] NCCL INFO Ring 00 : 0[0] -> 1[1] via P2P/direct pointer
ml:436:436 [1] NCCL INFO Ring 00 : 1[1] -> 2[2] via direct shared memory
ml:436:436 [2] NCCL INFO Ring 00 : 2[2] -> 3[3] via P2P/direct pointer
ml:436:436 [3] NCCL INFO Ring 00 : 3[3] -> 0[0] via direct shared memory
ml:436:436 [0] NCCL INFO Launch mode Group/Stream

CPU and memory utlization shows high usage and GPU utilization is 100%. Any thoughts? Thanks!

**Comment by wuhuikai** [comment]
`SyncBN` implemented in this repo does not support distributed training. All the GPUs should be in one machine.

**Comment by abhoi** [comment]
Hi @wuhuikai, all GPUs are on one machine in this case. We have 4 GPUs available. They are not on different machines. Any help?

**Comment by wuhuikai** [comment]
This issue is related to `Horovod docker`, which might be solved by switching to another `docker`.

**Comment by wuhuikai** [comment]
@abhoi The code in branch `latest` is now ready, which uses official `SyncBatchNorm` and can run on all OS with `PyTorch>=1.1.0`.
